fastavro/writer.py

```diff
@@ -202,7 +202,9 @@ def validate(datum, schema):
         return (
             isinstance(datum, Mapping) and
             all(
-                validate(datum.get(f['name']), f['type'])
+                validate(datum.get(f['name'], f.get('default')), f['type'])
+                if not (f['name'] not in datum and
```

Comment:
* I find this logic hard to read. Is there a way to create a utility function that will make this code clearer?
* Not sure if we want to fail. There [is a definition for type defaults](https://avro.apache.org/docs/1.8.0/spec.html#schema_record) even when there is no default value.